sdk/core/Azure.Core.TestFramework/src/Instrumentation/OperationInterceptor.cs

```diff
@@ -55,10 +55,17 @@ internal static object InvokeWaitForCompletion(object target, Type targetType, C
         {
             // get the concrete instance of OperationPoller.ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T>, TimeSpan?, CancellationToken)
             var poller = InjectZeroPoller();
-            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 3);
+            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 6);
```

Comment:
Here we invoke `OperationPoller.WaitForCompletionAsync(Operation operation, TimeSpan? timeSpan?, CancellationToken cancellationToken)` which will expand to the 6 parameter version. `UdpateStatusAsync`, `HasCompleted`, `Value`, `GetRawResponse` all come from the `operation` which we mock.
So here it should be OK to invoke this method? I roll back the codes and run the test locally. It seems fine.